changelog/@unreleased/pr-4558.v2.yml

```diff
@@ -0,0 +1,8 @@
+type: improvement
+improvement:
+  description: "In order to automatically use Nathan's optimization on single-column
+    table range scans, the generated code explicitly specifies the set of columns
+    in \"all columns\" range scan request. \n\nThis does not affect dynamic column
+    tables.\n"
```

Comment:
I think it's easier to say what the change does (as opposed to going into detail as to how we do it). So improvement is correct, but I'd write this more as something like:

> AtlasDB generated code now optimises `getRanges` and `getRangesLazy` to select only required columns. This is known to improve performance for single-column tables (Nathan's optimization). This does not affect dynamic column tables.

Also, I'd add at the end, for our non-metallic friends,

> AtlasDB users that check in their schemas need to regenerate them to benefit from this improvement.